src/hardware/vga_other.cpp

```diff
@@ -1218,84 +1122,99 @@ static void write_pcjr(io_port_t port, io_val_t value, io_width_t)
 	}
 }
 
-static uint8_t palette_num(const char *colour)
+void VGA_SetMonochromePalette(const enum MonochromePalette palette)
+{
+	if (machine == MCH_HERC) {
+		hercules_palette = palette;
+		VGA_SetHerculesPalette();
+
+	} else if (machine == MCH_CGA && mono_cga) {
+		mono_cga_palette = palette;
+		VGA_SetMonochromeCgaPalette();
+	}
+}
+
+static MonochromePalette cycle_forward(const MonochromePalette palette)
 {
-	if (strcasecmp(colour, "green") == 0)
-		return 0;
-	if (strcasecmp(colour, "amber") == 0)
-		return 1;
-	if (strcasecmp(colour, "white") == 0)
-		return 2;
-	if (strcasecmp(colour, "paperwhite") == 0)
-		return 3;
-	return 2;
+	auto value = (enum_val(palette) + 1) % (enum_val(MaxMonochromePalette) + 1);
+	return static_cast<MonochromePalette>(value);
 }
 
-void VGA_SetMonoPalette(const char *colour)
+static void cycle_mono_cga_palette(bool pressed)
 {
-	if (machine == MCH_HERC) {
-		herc_pal = palette_num(colour);
-		Herc_Palette();
+	if (!pressed) {
 		return;
 	}
-	if (machine == MCH_CGA && mono_cga) {
-		mono_cga_pal = palette_num(colour);
-		Mono_CGA_Palette();
+
+	mono_cga_palette = cycle_forward(mono_cga_palette);
+	VGA_SetMonochromeCgaPalette();
+
+	RENDER_SyncMonochromePaletteSetting(mono_cga_palette);
+}
+
+static void cycle_mono_cga_brightness(bool pressed)
+{
+	if (!pressed) {
 		return;
 	}
+	if (++mono_cga_bright > 1) {
+		mono_cga_bright = 0;
+	}
+	VGA_SetMonochromeCgaPalette();
 }
 
-static void CycleMonoCGAPal(bool pressed) {
-	if (!pressed) return;
-	if (++mono_cga_pal>3) mono_cga_pal=0;
-	Mono_CGA_Palette();
-}
+void VGA_SetMonochromeCgaPalette()
+{
+	constexpr auto num_cga_colors = 16;
```

Comment:
Might be able to programmatically get it with:

```c++
constexpr auto num_cga_colors = ARRAY_LEN(mono_cga_palettes[0]);
```

Reply:
I've introduced a constant for it.